.github/workflows/sc-app-deploy2.yml

```diff
@@ -1,48 +1,91 @@
+        description: "The branch prefix to use. Either feature|develop|release|hotfix|master"
+        required: true
+        default: ""
+      team-no:
+        description: "Team number. Always required."
+        required: true
+        default: ""
+      ticket-id:
+        description: "Ticket ID on feature or hotfix branch. Version number on master and release branch"
+        required: false
+        default: ""
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
-  build:
+  # This workflow contains a single job called "deploy"
+  deploy:
     # The type of runner that the job will run on
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
+    env:
+      # Setting environment variables here make them available for all steps
+      CI_GITHUB_TRAVISUSER_SWARMVM_KEY: ${{ secrets.CI_GITHUB_TRAVISUSER_SWARMVM_KEY }}
+      DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+      DOCKER_TOKEN: ${{ secrets.DOCKER_TOKEN }}
+    defaults:
+      # Setting defaults for all steps, this require the first action must check-out the code
+      run:
+        working-directory: ./sc-app-deploy
 
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it must be on first place to allow central setting of working directory
       - uses: actions/checkout@v2
-
-      # Print the available Python version information
-      - name: Python environment
+      - name: Echo parameter
         run: |
-          import os
-          import sys
-          print("Python version")
-          print (sys.version)
-          print("Version info.")
-          print (sys.version_info)          
-          print(os.environ['PATH'])
-        shell: python
-
-      # Print the working directory and allow the execution of the sc-app-deploy.py script.
+          echo "GITHUB event trigger:       ${{ github.event_name }}
+          echo "GITHUB input branch-prefix: ${{ github.event.inputs.branch-prefix }}
+          echo "GITHUB input team-no:       ${{ github.event.inputs.team-no }}
+          echo "GITHUB input ticket-id:     ${{ github.event.inputs.ticket-id }}
+
       - name: Display working directory
+        # Just for failure analysis to track down possible issues occurring for invalid host configurations
         run: |
           ls -la .
-          chmod a+x ./sc-app-deploy/sc-app-deploy.py
+          env
+          chmod a+x ./sc-app-deploy.py
         shell: bash
+      # Run the deployment based on the arguments given
+      - name: Running scheduled deployment for develop
+        if: github.event_name == 'schedule'
+        # 9999 as team number is a dummy to overcom argparse limitations
+        run: ./sc-app-deploy.py develop --scheduled 9999
```

Comment:
1. Actually it is a limitation of using positional parameters. Would be no problem with a keyword parameter.

2. What "scheduled" is doing, is to deploy the branch to the test host. So I would name it that way. Use the provider perspective, not the caller perspective. 

So for the develop "command" I would suggest:
... develop [--deployhost {"test"|"team"}] [--teamnumber <teamnumber>]

Examples:
... develop --deployhost test
... develop --teamnumber 1  << Defaults to the teamhost
... develop --deployhost team --teamnumber 1

3. It is unintuitive that the branches are represented by a subcommand here. Why not just using a "branchtype" parameter? 
Currently: sc-app-deploy.py develop --scheduled 9999
Proposed: sc-app-deploy.py --branchtype develop --deployhost test

Currently: sc-app-deploy.py develop 1 2312
Proposed: sc-app-deploy.py --branchtype develop --teamnumber 1 --jiraid 2312

I think this is much more the standard way and leaves room for real additional subcommands. E.g. if we want to have some kind of deployment cleanup or something similar.

Reply:
Parameter parsing changed to non positional parameters